src/Calypso-SystemTools-Core/ClyMethodEditorCommand.class.st

```diff
@@ -17,9 +17,22 @@ Class {
 	#category : #'Calypso-SystemTools-Core-Editors-Methods'
 }
 
+{ #category : #testing }
+ClyMethodEditorCommand class >> canBeExecutedInCodeMethodEditor: aTool [
+	
+	^ false
+]
+
 { #category : #testing }
 ClyMethodEditorCommand class >> canBeExecutedInContext: aToolContext [
-	^aToolContext tool isKindOf: ClyMethodEditorToolMorph 
+
+	^ aToolContext tool isCommandAvailable: self
+]
+
+{ #category : #testing }
+ClyMethodEditorCommand class >> canBeExecutedInMethodEditor: aTool [
+
+	^ false
```

Comment:
It should be true by default

Reply:
no it shouldn't. If you put true by default you will have the same problem as before.

Reply:
This method is related to main superclass of method editors ClyMethodEditorTool.
And all subclasses of ClyMethodEditorCommand are supposed to work with any such tools:
- you can edit tag and package in existing method editor and in new method editor.

Reply:
Maybe I not understand your intention. Do you want to disable command in "existing method" editor? (and keep it in method creation tool)

Reply:
since it does not works, yes.

Reply:
but when #defaultMenuItemName is called methodEditor is nil, so I do not have a way to know where it is going it act (as for change the name).